### PR TITLE
Fix the problem of updating categories and menus in batch

### PR DESCRIPTION
**Pull request by guqing**
## What this PR does?
修复批量保存方法会刷新缺省字段为`null`的问题
see #1521

### CODE REVIEW

File: src/main/java/run/halo/app/service/impl/CategoryServiceImpl.java [hunk 1]
```diff
@@ -661,15 +662,21 @@ private boolean doCategoryHasEncrypt(
...
-        for (Category category : categories) {
-            resultList.add(update(category));
-        }
-        return resultList;
+        Set<Integer> categoryIds = ServiceUtils.fetchProperty(categories, Category::getId);
+        Map<Integer, Category> idCategoryParamMap =
+            ServiceUtils.convertToMap(categories, Category::getId);
+        return categoryRepository.findAllById(categoryIds)
+            .stream()
+            .map(categoryToUse -> {
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/service/impl/CategoryServiceImpl.java:675]: I suggesting naming `categoryToUse` to `categoryToUpdate`.

**Review by JohnNiang — CHANGES_REQUESTED** [review]
Nice work. I've left a comment below, please take some moment to look at it.

File: src/main/java/run/halo/app/service/impl/CategoryServiceImpl.java (hunk 1 above)
Comment (guqing) [reply src/main/java/run/halo/app/service/impl/CategoryServiceImpl.java:675]: ok

**Review by JohnNiang — APPROVED** [review]
LGTM

**Comment by JohnNiang** [comment]
I'm going to merge this PR due to no negative feedback.

**Pull request merged**